lib/virtual-io/src/selector.rs

```diff
@@ -1,51 +1,62 @@
 use std::{
-    collections::HashSet,
-    mem::ManuallyDrop,
-    sync::{
-        mpsc::{Receiver, Sender},
-        Arc, Mutex,
-    },
+    collections::HashMap,
+    io,
+    sync::{Arc, Mutex},
 };
 
 use derivative::Derivative;
-use mio::Token;
+use mio::{Registry, Token};
 
-use crate::{HandlerWrapper, InterestType};
+use crate::{InterestHandler, InterestType};
 
 #[derive(Derivative)]
 #[derivative(Debug)]
 pub(crate) struct EngineInner {
+    seed: usize,
+    registry: Registry,
     #[derivative(Debug = "ignore")]
-    selector: mio::Poll,
-    rx_drop: Receiver<Token>,
+    lookup: HashMap<Token, Box<dyn InterestHandler + Send + Sync>>,
 }
 
 #[derive(Derivative)]
 #[derivative(Debug)]
 pub struct Selector {
+    token_close: Token,
     inner: Mutex<EngineInner>,
-    #[derivative(Debug = "ignore")]
-    pub(crate) registry: mio::Registry,
-    pub(crate) tx_drop: Mutex<Sender<Token>>,
     closer: mio::Waker,
 }
 
+use lazy_static::lazy_static;
+lazy_static! {
+    static ref GLOBAL_SELECTOR: Arc<Selector> = Selector::new_internal();
```

Comment:
Can we document why there is a global selector, and what it is used for?

Can this be a contention point in highly concurrent scenarios? (eg Edge)

Reply:
deleted it